hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,313 @@
+---
+title: "PWM HWIP Technical Specification"
+---
+
+# Overview
+
+This document specifies PWM hardware IP (HWIP) functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification/index.md" >}})
+See that document for integration overview within the broader top-level system.
+
+## Features
+
+- Pulse-width modulated (PWM) with adjustable duty cycle
+- Suitable for general-purpose use, but primarily designed for control of tri-color LEDs
+- Parametrizable number of output channels
+- Separate clock domains for TL-UL I/O vs. core operations
+   - PWM operation can continue in low-power state.
+- Independent control of duty cycle, phase, and polarity for all channels
+- Programmable resolution (1 to 16 bits) in adjusting the output duty cycle and phase
+- All output channels are driven at the same pulse frequency, which is selected by a 27-bit clock divider
+- Hardware-controlled blink feature
+   - Blinking channel toggles between two separably programmable duty cycles
+   - Blink timing is independently programmable for each channel
+- "Heartbeat" blink mode
+   - Output duty cycle linearly increments and decrements between two programmable values
+   - Step size and step frequency are independently programmable for each channel
+- Channels can be configured to blink synchronously or independently
+- All duty cycle settings are independently programmable
+
+## Description
+
+The PWM IP is primarily designed to drive a parametrizable number of pulse-width modulated outputs with periodic pulses each with a programmable frequency, phase, and duty cycle (i.e. the ratio between the pulse duration to the overall period between pulses.)
```

Comment:
Nit: Should be the ratio between A *and* B.

Reply:
Got it.